### DISCUSSION TITLE
useRequest cacheKey not working.

### DISCUSSION THREAD

**Discussion by xXAvoraXx** [umijs/umi#11407] [Q&A]
使用useRequest時，緩存機制不起作用，但是當我使用最新版本的ahooks request時，緩存起作用。 如何進行手動升級?
With useRequest(ahooks2.0), the cache mechanism does not work, but when I use the latest version(ahooks3.7.7) ahooks request, the cache works. How can I do a manual upgrade?

https://umijs.org/docs/max/request
> 需要注意的是，ahooks 已经更新到3.0，而我们为了让 umi@3 的项目升级起来不那么困难，继续沿用了 ahooks2.0

## Context

- **Umi Version**: "@umijs/max": "^4.0.72",
- **Node Version**: "node": ">=12.0.0"
- **Platform**: Windows 11, Chrome

**Comment by scplay** [comment]
我也发现设置 cacheKey + staleTime 完全不生效，调用 run 还是会发出请求，看了下 umijs/max 使用的是 ahooks/use-request v2.8, 直接下载 ahooks 的 v3.8 版本使用这两个参数是没有问题，不知道哪里出了问题，有时间再看看源码

**Reply by xXAvoraXx** [reply]
Instead of dealing with this problem, I started using tanstack react-query.
